hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
For reference, here's the same data with the threshold changed to `n < 100`. I almost think this is more useful.

Here's a link so I'm not spamming the PR with tons of images:
https://imgur.com/a/SFrTl